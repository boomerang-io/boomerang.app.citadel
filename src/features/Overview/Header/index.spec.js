import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Header } from "./index";

const mockfn = jest.fn();
const teams = [
  {
    "id": "5a8b331e262a70306622df73",
    "name": "ATT",
    "higherLevelGroupId": "59aebcf47424530fce952fa2",
    "ucdApplicationId": "96c22f37-5443-4bf5-83bb-7ce90deec820",
    "isActive": true,
    "audits": [
      {
        "auditerId": null,
        "date": 1519072030853,
        "note": "CI-Team created"
      }
    ],
    "boomerangTeamName": "AT&T MIL Mobile@Scale",
    "boomerangTeamShortname": "ms-att-mil"
  }
];

describe("Header --- Snapshot", () => {
  it("Capturing Snapshot of Header", () => {
    const renderedValue = renderer
      .create(
        <Header
          handleChangeTeam={mockfn}
          teams={teams}
        />
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("Header --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Header
        handleChangeTeam={mockfn}
        teams={teams}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
