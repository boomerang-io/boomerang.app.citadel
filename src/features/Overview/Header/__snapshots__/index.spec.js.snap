// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header --- Snapshot Capturing Snapshot of Header 1`] = `
<div
  className="container"
>
  <h2
    className="title"
  >
    Team
  </h2>
  <div
    aria-label="Choose an item"
    className="bx--form-item bx--combo-box bx--list-box"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="open menu"
      className="bx--list-box__field"
      data-toggle={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-label="ListBox input field"
        autoComplete="off"
        className="bx--text-input"
        disabled={false}
        id="downshift-0-input"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a team"
        role="combobox"
        selectedItem={null}
        title="Team"
        value=""
      />
      
      <div
        className="bx--list-box__menu-icon"
      >
        <svg
          alt="Open menu"
          aria-label="Open menu"
          fillRule="evenodd"
          focusable="false"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
